extensions/nns/src/install_nns.rs

```diff
@@ -483,6 +486,60 @@ pub fn set_cmc_authorized_subnets(
         .map_err(|e| anyhow!("Call to propose to set authorized subnets failed: {e}"))
 }
 
+/// Sets the cycles ledger canister id in the CMC
+#[context("Failed to set the cycles ledger canister id in the CMC")]
+pub fn set_cycles_ledger_canister_id_in_cmc(
+    nns_url: &Url,
+    dfx_cache_path: &Path,
+) -> anyhow::Result<()> {
+    #[derive(CandidType, Clone, Debug, PartialEq, Eq)]
+    struct CyclesCanisterInitPayload {
+        pub cycles_ledger_canister_id: Option<Principal>,
+    }
+
+    let wasm_path = nns_wasm_dir(dfx_cache_path);
+    let ledger_wasm_path = wasm_path.join(NNS_CYCLES_MINTING.wasm_name);
+    let ledger_wasm_bytes = dfx_core::fs::read(&ledger_wasm_path)?;
+    let wasm_hash = Sha256::digest(ledger_wasm_bytes);
+    let mut upgrade_arg_file = tempfile::NamedTempFile::new()?;
+    upgrade_arg_file
+        .write_all(
+            &Encode!(&(Some(CyclesCanisterInitPayload {
+                cycles_ledger_canister_id: Some(
+                    Principal::from_text(CYCLES_LEDGER.canister_id).unwrap()
+                ),
+            }),))
+            .unwrap(),
+        )
+        .unwrap();
```

Comment:
the unwrap on Encode!() I can go along with, .write_all().unwrap() probably not

Reply:
Agreed, and there were more mistakes. I must have dropped the self-review somehow